File: internal/app/dto/address.go
package dto

type CustomerAddressDTO struct {
	ID                uint   `json:"id"`
	CustomerProfileID uint   `json:"customerProfileId"`
	Label             string `json:"label"`
	RecipientName     string `json:"recipientName"`
	Phone             string `json:"phone"`
	Country           string `json:"country"`
	Province          string `json:"province"`
	City              string `json:"city"`
	District          string `json:"district"`
	AddressLine1      string `json:"addressLine1"`
	AddressLine2      string `json:"addressLine2"`
	PostalCode        string `json:"postalCode"`
	IsDefault         bool   `json:"isDefault"`
	IsTest            bool   `json:"isTest"`
	ValidationStatus  string `json:"validationStatus"`
	ValidationDetail  string `json:"validationDetail"`
	ExtraData         string `json:"extraData"`
	CreatedAt         string `json:"createdAt"`
	UpdatedAt         string `json:"updatedAt"`
}

type CreateAddressInput struct {
	CustomerProfileID uint   `json:"customerProfileId"`
	Label             string `json:"label"`
	RecipientName     string `json:"recipientName"`
	Phone             string `json:"phone"`
	Country           string `json:"country"`
	Province          string `json:"province"`
	City              string `json:"city"`
	District          string `json:"district"`
	AddressLine1      string `json:"addressLine1"`
	AddressLine2      string `json:"addressLine2"`
	PostalCode        string `json:"postalCode"`
	IsDefault         bool   `json:"isDefault"`
	IsTest            bool   `json:"isTest"`
	ValidationStatus  string `json:"validationStatus"`
	ValidationDetail  string `json:"validationDetail"`
	ExtraData         string `json:"extraData"`
}

type UpdateAddressInput struct {
	ID                uint   `json:"id"`
	CustomerProfileID uint   `json:"customerProfileId"`
	Label             string `json:"label"`
	RecipientName     string `json:"recipientName"`
	Phone             string `json:"phone"`
	Country           string `json:"country"`
	Province          string `json:"province"`
	City              string `json:"city"`
	District          string `json:"district"`
	AddressLine1      string `json:"addressLine1"`
	AddressLine2      string `json:"addressLine2"`
	PostalCode        string `json:"postalCode"`
	IsDefault         bool   `json:"isDefault"`
	IsTest            bool   `json:"isTest"`
	ValidationStatus  string `json:"validationStatus"`
	ValidationDetail  string `json:"validationDetail"`
	ExtraData         string `json:"extraData"`
}

type BindAddressInput struct {
	FulfillmentLineID uint `json:"fulfillmentLineId"`
	CustomerAddressID uint `json:"customerAddressId"`
}

File: frontend/src/entities/address.ts
export interface CustomerAddressDTO {
  id: number
  customerProfileId: number
  label: string
  recipientName: string
  phone: string
  country: string
  province: string
  city: string
  district: string
  addressLine1: string
  addressLine2: string
  postalCode: string
  isDefault: boolean
  isTest: boolean
  validationStatus: string
  validationDetail: string
  extraData: string
  createdAt: string
  updatedAt: string
}

export interface CreateAddressInput {
  customerProfileId: number
  label: string
  recipientName: string
  phone: string
  country: string
  province: string
  city: string
  district: string
  addressLine1: string
  addressLine2: string
  postalCode: string
  isDefault: boolean
  isTest: boolean
  validationStatus: string
  validationDetail: string
  extraData: string
}

export interface UpdateAddressInput {
  id: number
  customerProfileId: number
  label: string
  recipientName: string
  phone: string
  country: string
  province: string
  city: string
  district: string
  addressLine1: string
  addressLine2: string
  postalCode: string
  isDefault: boolean
  isTest: boolean
  validationStatus: string
  validationDetail: string
  extraData: string
}

export interface BindAddressInput {
  fulfillmentLineId: number
  customerAddressId: number
}
